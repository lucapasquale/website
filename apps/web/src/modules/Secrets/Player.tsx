'use client'
import ReactPlayer from 'react-player'

type Props = {
  playlistURL: string
}

export function Player({ playlistURL }: Props) {
  return (
    <div className="hidden mt-8 sm:flex sm:justify-center sm:w-screen">
      <ReactPlayer controls url={playlistURL} />
    </div>
  )
}
