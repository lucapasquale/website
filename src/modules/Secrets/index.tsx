import ReactPlayer from "react-player";

import { Hero } from "~components/Hero";
import { Link } from "~components/Link";

const PLAYLIST_LINK = "https://www.youtube.com/playlist?list=PL2gDVp_0vZOQjqMex201dYpUiu1mcGX96";

export function Page() {
  return (
    <>
      <Hero title="Secrets" subTitle="How did you get here?" />

      <article className="flex flex-col items-center px-8">
        <Link to={PLAYLIST_LINK}>My dumb playlist</Link>

        <div className="hidden mt-8 sm:flex sm:justify-center sm:w-screen">
          <ReactPlayer controls url={PLAYLIST_LINK} />
        </div>
      </article>
    </>
  );
}
