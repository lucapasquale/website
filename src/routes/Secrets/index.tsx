import React, { FC } from 'react'
import Helmet from 'react-helmet'
import ReactPlayer from 'react-player'
import styled from 'styled-components'

import { Hero } from '../../components/Hero'
import { Layout } from '../../components/Layout'

const Section = styled.section({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  padding: '32px',
})

const Player = styled.div({
  marginTop: '32px',

  '@media (max-width: 700px)': {
    display: 'none',
  }
})

const PLAYLIST_LINK =
  'https://www.youtube.com/playlist?list=PL2gDVp_0vZOQjqMex201dYpUiu1mcGX96'

export const Secrets: FC = () => (
  <>
    <Helmet>
      <title>Luca Pasquale - Secrets</title>
      <link rel="canonical" href="https://lucapasquale.github.io/secrets" />
    </Helmet>

    <Layout>
      <Hero
        title="Secrets"
        subTitle="How did you get here?"
      />

      <Section>
        <a target="_blank" href={PLAYLIST_LINK}>
          My dumb playlist
        </a>

        <Player>
          <ReactPlayer controls url={PLAYLIST_LINK} />
        </Player>
      </Section>
    </Layout>
  </>
)
