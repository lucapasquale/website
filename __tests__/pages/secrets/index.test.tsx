import { render, screen, waitFor, within } from '@testing-library/react'

import Secrets from '../../../src/pages/secrets'

describe('Secrets', () => {
  it('snapshot', () => {
    const { container } = render(<Secrets />)

    expect(container).toMatchSnapshot()
  })

  it('renders the page', async () => {
    render(<Secrets />)

    expect(screen.getByRole('link')).toBeInTheDocument()
  })
})
