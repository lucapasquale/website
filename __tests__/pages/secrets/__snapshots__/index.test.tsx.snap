// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Secrets snapshot 1`] = `
<div>
  <header
    class="mb-12"
  >
    <h1
      class="text-4xl font-bold pb-2"
    >
      Secrets
    </h1>
    <h2
      class="text-gray-400"
    >
      How did you get here?
    </h2>
  </header>
  <article
    class="flex flex-col items-center px-8"
  >
    <a
      href="https://www.youtube.com/playlist?list=PL2gDVp_0vZOQjqMex201dYpUiu1mcGX96"
      rel="noopener noreferrer"
      target="_blank"
    >
      My dumb playlist
    </a>
    <div
      class="hidden mt-8 sm:flex sm:justify-center sm:w-screen"
    />
  </article>
</div>
`;
